### PR TITLE
[Shopify] Add Delete Customers implementation for Shopify

### LINKED ISSUE #7762: [Shopify] Implement DELETE for the Customers table
MindsDB integrates with Shopify, so you can fetch data from your online store and use it to make relevant predictions and forecasts with MindsDB.

## Resources

Familiarize yourself with the existing Shopify integration: https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/shopify_handler

## Steps to Follow

Currently, the Shopify integration with MindsDB implements SELECT from the Customers table.

The task is to extend [this implementation](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/shopify_handler) to include the DELETE command to delete data from the Customers table.

## :loudspeaker: Additional rewards :trophy:

Each integration extension brings 15+ points for SWAG and entry into the draw for a :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)

:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by dalalvarun**
## Description
Added the implementation to delete customers by interacting with Shopify API

Fixes #7762 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.
   -  Go to mindsdb online editor
   - Execute the query "DELETE FROM shopify_datasource.customers WHERE first_name = 'some_first_name';"
   - The query will get executed without any ImplementationError.
   - Verify that the item has been deleted by checking the items returned by the query "SELECT * from shopify_datasource ; "

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

https://github.com/mindsdb/mindsdb/assets/70593335/34af8060-dd74-4d88-b47a-e933f7ce612a

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by dalalvarun** [comment]
@martyna-mindsdb Please review my first PR😄

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by dalalvarun** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dalalvarun** [comment]
recheck

**Comment by martyna-mindsdb** [comment]
@dalalvarun 
Please provide a screenshot or a short video presenting the DELETE works fine.

**Comment by dalalvarun** [comment]
> @dalalvarun Please provide a screenshot or a short video presenting the DELETE works fine.

Added to the description!

**Comment by martyna-mindsdb** [comment]
Thanks @dalalvarun 

Also, please add a sample DELETE command to the README file.

**Comment by dalalvarun** [comment]
Done @martyna-mindsdb

**Review by martyna-mindsdb — APPROVED** [review]

**Comment by tomhuds** [comment]
@martyna-mindsdb please ensure tests have been added, and the checklist has been completed

**Comment by martyna-mindsdb** [comment]
@tomhuds 
@ZoranPandovski and I are following a similar pattern when merging this kind of PRs (that add one functionality to a handler).
Contributors do not write tests for those, like in [this PR](https://github.com/mindsdb/mindsdb/pull/7666).

Please specify what you mean by tests and checklist.

**Comment by tomhuds** [comment]
Wouldn't it be great if they did write tests? We ask for tests from the core team.

By checklist I mean the PR checklist, i.e. 
image
The purpose of the checklist is to ensure all items are completed before the PR is merged

**Comment by dalalvarun** [comment]
Hi @martyna-mindsdb @tomhuds @ZoranPandovski 
I see that the Shopify integration didnt have unit tests since the beginning. I can add it for you. Can you please create an issue for this and assign it to me?

**Comment by martyna-mindsdb** [comment]
@dalalvarun 
Here it is: https://github.com/mindsdb/mindsdb/issues/7890

**Pull request merged**
